components/pages/UserStories/UserStories.js

```diff
@@ -226,16 +227,21 @@ export function UserStories({ userId }) {
                       ))}
                     </div>
                   )}
+
                   <Avatar
                     user={user}
                     size={activeUser ? 'small' : 'medium'}
                     className={cx({
                       [styles.storyAvatarSmall]: !activeUser,
                       [styles.storyAvatarActive]: activeUser,
                     })}
-                    colorRing={!activeUser}
+                    colorRing={
+                      !activeUser && !prevUserName.includes(user.user_name)
```

Comment:
You already have progress info for each user stored in `storyIndices`. `storyIndices` is a map of `userId => number of stories watched`, if that number is equal to number of stories available, it means the user is finished.

```suggestion
                      !activeUser && storyIndices[user.user_id] === user.stories.length - 1
```